.github/workflows/python-max-version.yml

```diff
@@ -0,0 +1,71 @@
+name: python-max-version
+on: [push]
+jobs:
+  checkout:
+    runs-on: ubuntu-latest
+    container: circleci/python:3.10.0
+    env:
+      SUDO: 'sudo'
+    steps:
+      # This Docker file changes sets USER to circleci instead of using the default user, so we need to update file permissions for this image to work on GH Actions.
+      # See https://docs.github.com/actions/reference/virtual-environments-for-github-hosted-runners#docker-container-filesystem
+      - name: Setup file system permissions
+        run: sudo chmod -R 777 $GITHUB_WORKSPACE /github /__w/_temp
+      - name: Checkout Streamlit code
+        uses: actions/checkout@v3
+      - name: "update submodules"
+        run: |
+          git submodule init
+          git submodule update --remote
+        shell: bash
+      - name: "Install NVM, Node.js, and Yarn"
+        uses: actions/setup-node@v3
+        with:
+          node-version: lts/gallium
+          cache: 'npm'
+          cache-dependency-path: frontend/package-lock.json
+      - name: Install pyodbc dependencies
+        run: |
+          ${SUDO} apt-get install -y unixodbc-dev
+        shell: bash
+      - name: Install graphviz dependencies
+        run: |
+          ${SUDO} apt update
+          ${SUDO} apt-get install -y libgvc6
+        shell: bash
+      - name: create virtualenv
+        run: |
+          echo 'Checking for virtualenv'
+          if [ ! -d venv ] ; then
+            # The virtualenv was NOT restored from cache. Create a new one.
+            python -m venv venv
+            source venv/bin/activate
+            pip install --upgrade pip
+            make setup
+            make pipenv-install
+            deactivate
+          else
+            # The virtualenv WAS restored from cache. Don't create a new one.
+            echo 'Virtualenv already exists, not creating'
+          fi
+
+        shell: bash
+      - name: generate_protobufs
+        run: |
+          # install protobuf v3
+          ${SUDO} apt update
+          ${SUDO} apt-get install -y gnupg
+          ${SUDO} pip install mypy-protobuf
```

Comment:
This command installs `protobuf` 4+ which is not compatible with `streamlit`. It is worth adding an explicit restriction here, so as not to download incompatible versions of libraries on the CI.